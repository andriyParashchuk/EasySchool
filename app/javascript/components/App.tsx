import * as React from 'react'
import * as ReactDOM from 'react-dom'

interface AppProps {
  name: string,
}

export class App extends React.Component<AppProps> {
  render() {
    const { name } = this.props
    return <div>Hello {name}!</div>
  }
}
