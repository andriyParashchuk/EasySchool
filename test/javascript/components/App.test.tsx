import * as React from 'react'
import { shallow } from 'enzyme'

import { App } from '../../../app/javascript/components/App'

it('Renders component with props', () => {
  const name = "Test"

  const component = shallow(
    <App
      name={name}
    />,
  )

  expect(component).toMatchSnapshot()
  expect(component.matchesElement(<div>Hello Test!</div>)).toBeTruthy()
})
