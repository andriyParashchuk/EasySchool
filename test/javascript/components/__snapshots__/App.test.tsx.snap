// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders component with props 1`] = `
<div>
  Hello 
  Test
  !
</div>
`;
